File: tictactoe.py
import random
board = ['_', '_', '_', '_', '_', '_', '_', '_', '_', '_']

def addX():
    a = int(input("X to Enter position(1-9):"))
    if(board[a] == '_'):
        board[a] = 'X'
    else:
        print("Position already occupied!!\a")
        addX()

def addO():
    a = int(input("O to Enter position(1-9):"))
    if(board[a] == '_'):
        board[a] = 'O'
    else:
        print("Position already occupied!!\a")
        addO()

def checkwin():#return 1 = X win, return 2 = O win
    if(board[1] == board[2] == board[3] != '_'):
        if(board[1] == 'X'):
            return 1
        else:
            return 2
    elif(board[4] == board[5] == board[6] != '_'):
        if(board[4] == 'X'):
            return 1
        else:
            return 2
    elif(board[7] == board[8] == board[9] != '_'):
        if(board[7] == 'X'):
            return 1
        else:
            return 2
    elif(board[1] == board[4] == board[7] != '_'):
        if(board[1] == 'X'):
            return 1
        else:
            return 2
    elif(board[2] == board[5] == board[8] != '_'):
        if(board[2] == 'X'):
            return 1
        else:
            return 2
    elif(board[3] == board[6] == board[9] != '_'):
        if(board[3] == 'X'):
            return 1
        else:
            return 2
    elif(board[1] == board[5] == board[9] != '_'):
        if(board[1] == 'X'):
            return 1
        else:
            return 2
    elif(board[3] == board[5] == board[7] != '_'):
        if(board[3] == 'X'):
            return 1
        else:
            return 2
    else:
        return 0

def printboard():
    print(board[1], board[2], board[3])
    print(board[4], board[5], board[6])
    print(board[7], board[8], board[9])

a = input("Name of player 1:")
b = input("Name of player 2:")
c = random.randint(1,2)
if(c == 1):
    print(a, "is X")
    print(b, "is O")
else:
    print(a, "is O")
    print(b, "is X")
printboard()

while True:
    addX()
    printboard()
    d = checkwin()
    if(d == 1):
        print("X wins!!")
        exit()
    elif(d == 2):
        print("O wins!!")
        exit()
    addO()
    printboard()
    d = checkwin()
    if(d == 1):
        print("X wins!!")
        exit()
    elif(d == 2):
        print("O wins!!")
        exit()

File: tictactoe.c
#include <stdio.h>
#include <stdlib.h>

#define SIZE 3

void clearScreen() {
    #ifdef _WIN32
        system("cls");
    #else
        system("clear");
    #endif
}

void printBoard(char board[SIZE][SIZE]) {
    printf("\n");
    for (int i = 0; i < SIZE; i++) {
        for (int j = 0; j < SIZE; j++) {
            printf(" %c ", board[i][j]);
            if (j < SIZE - 1) printf("|");
        }
        printf("\n");
        if (i < SIZE - 1) printf("-----------\n");
    }
    printf("\n");
}

int checkWin(char board[SIZE][SIZE], char player) {
    for (int i = 0; i < SIZE; i++) {
        if (board[i][0] == player && board[i][1] == player && board[i][2] == player) return 1;
        if (board[0][i] == player && board[1][i] == player && board[2][i] == player) return 1;
    }
    if (board[0][0] == player && board[1][1] == player && board[2][2] == player) return 1;
    if (board[0][2] == player && board[1][1] == player && board[2][0] == player) return 1;
    return 0;
}

int checkDraw(char board[SIZE][SIZE]) {
    for (int i = 0; i < SIZE; i++) {
        for (int j = 0; j < SIZE; j++) {
            if (board[i][j] == ' ') return 0;
        }
    }
    return 1;
}

int main() {
    char board[SIZE][SIZE] = {
        {' ', ' ', ' '},
        {' ', ' ', ' '},
        {' ', ' ', ' '}
    };
    int row, col;
    char player = 'X';
    int moveCount = 0;

    while (1) {
        clearScreen();
        printBoard(board);

        printf("Player %c's turn. Enter row (1-3) and column (1-3): ", player);
        scanf("%d %d", &row, &col);
        row--; col--;

        if (row < 0 || row >= SIZE || col < 0 || col >= SIZE || board[row][col] != ' ') {
            printf("Invalid move! Try again.\n");
            continue;
        }

        board[row][col] = player;
        moveCount++;

        if (checkWin(board, player)) {
            clearScreen();
            printBoard(board);
            printf("Player %c wins!\n", player);
            break;
        }

        if (checkDraw(board)) {
            clearScreen();
            printBoard(board);
            printf("It's a draw!\n");
            break;
        }

        player = (player == 'X') ? 'O' : 'X';
    }

    return 0;
}
